src/components/posts-index.js

```diff
@@ -0,0 +1,100 @@
+
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchWeather, setDefault } from '../actions';
+import { useForm } from "react-hook-form";
+import _ from 'lodash';
+import { useEffect } from 'react';
+import { Sparklines, SparklinesReferenceLine, SparklinesLine } from 'react-sparklines';
+
+const WeatherIndex = () => {
```

Comment:
This is a small app, but this component is getting large. What ways could you make it smaller?